llvm/lib/Transforms/Yk/StackMaps.cpp

```diff
@@ -60,8 +60,15 @@ class YkStackmaps : public ModulePass {
             // We don't need to insert stackmaps after intrinsics. But since we
             // can't tell if an indirect call is an intrinsic at compile time,
             // emit a stackmap in those cases too.
-            if (!CI.isIndirectCall() && CI.getCalledFunction()->isIntrinsic())
+            
+            if (!CI.isIndirectCall() &&
+                (CI.getCalledFunction()->isIntrinsic() ||
+                 (CI.getCalledFunction()->isDeclaration() &&
+                  (!CI.getCalledFunction()->getName().startswith(
+                       "yk_promote") &&
+                   CI.getCalledFunction()->getName() != YK_NEW_CONTROL_POINT))))
```

Comment:
I still have a feeling we might get away with omitting stackmaps for the control point too, but we can investigate that in a separate PR. It will require some changes to yk too.